doc/network.md

```diff
@@ -0,0 +1,246 @@
+
+#### `plasmid:workspace/new`
+
+Sent by the client to request the server to create a new workspace. 
+
+Direction: `C->S`
+
+| Fields    | Type           | Description                                                            |
+|:---------:|:--------------:|:-----------------------------------------------------------------------|
+| workspace | Identifier     | The identifier of the new workspace.                                   |
+| bounds    | [Bounds]       | The bounds of the map, may be `[[0, 0, 0], [0, 0, 0]]` if not present. |
+| generator | string (32767) | The generator string for the map, may be empty if unspecified.         |
+| has_data  | bool           | `true` if contains a data NBT tag, else `false`.                       |
+| data      | (NBT Tag)?     | Arbitrary data assigned to the map template.                           |
+
+#### `plasmid:workspace/leave`
+
+Leaves the specified workspace.
+
+Direction: `C<->S`
+
+| Fields    | Type       | Description                           |
+|:---------:|:----------:|:--------------------------------------|
+| workspace | Identifier | The identifier of the left workspace. |
+
+#### `plasmid:workspace/bounds`
+
+Sets the bounds of the workspace map. The client should render those bounds if non-null.
+A client can request bounds change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction `C<->S`
+
+| Fields    | Type       | Description                                                                      |
+|:---------:|:----------:|:---------------------------------------------------------------------------------|
+| workspace | Identifier | The identifier of the affected workspace.                                        |
+| bounds    | [Bounds]   | The new bounds of the workspace map, `[[0, 0, 0], [0, 0, 0]]` if bounds is null. |
+
+#### `plasmid:workspace/data`
+
+Sets the associated data on the workspace map.
+A client can request data change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction: `C<->S`
+
+| Fields    | Type       | Description                               |
+|:---------:|:----------:|:------------------------------------------|
+| workspace | Identifier | The identifier of the affected workspace. |
+| data      | NBT Tag    | The NBT data.                             |
+
+#### `plasmid:workspace/region`
+
+Specifies a single region, the client should render a box, and a marker tag for it.
+If sent from a client, this is considered as an update request which can be rejected.
+
+Direction: `C<->S`
+
+| Fields     | Type       | Description                                                          |
+|:----------:|:----------:|:---------------------------------------------------------------------|
+| workspace  | Identifier | The identifier of the workspace in which the region should be added. |
```

Comment:
With all these packets we can probably exclude the workspace identifier and force the assumption that it is the current workspace. This helps with network traffic as well as implicit security in the C->S direction.